shared/domains/chelonia/utils.js

```diff
@@ -686,7 +688,11 @@ export function eventsAfter (contractID: string, sinceHeight: number, limit?: nu
                   const hash = GIMessage.deserializeHEAD(currentEvent).hash
                   const height = GIMessage.deserializeHEAD(currentEvent).head.height
                   if (height !== sinceHeight || (sinceHash && sinceHash !== hash)) {
-                    controller.error(new Error('hash() !== since'))
+                    if (height === sinceHeight && sinceHash && sinceHash !== hash) {
+                      controller.error(new ChelErrorForkedChain('Forked chain: hash() !== since'))
+                    } else {
+                      controller.error(new Error('hash() !== since'))
+                    }
```

Comment:
Why are we just logging the error and not throwing it?

Also, it's possible `hash() !== since` could be logged in other places in the future too so to avoid ambiguity, could the `console.error` messages please be updated to make it clear where this is happening?


Also, above on line 681 we have a comment that's inconsistent with the code:

```js
// Check if the event limit is reached; if so, throw an error
```

No error is thrown.

Reply:
This isn't logging anything, it's in fact throwing an error.

See <https://developer.mozilla.org/en-US/docs/Web/API/WritableStreamDefaultController/error>

Reply:
Oh, I misread `controller` as `console` 🤦 

What about this comment from that link you shared?

```
This method is rarely used, since usually it suffices to return a rejected promise from one of the underlying sink's methods. However, it can be useful for suddenly shutting down a stream in response to an event outside the normal lifecycle of interactions with the underlying sink.
```

What's the reason you chose to do it this way as opposed to `return a rejected promise`?

Reply:
ETA: Maybe you misread `console` instead of `controller`?